tools/lsmod.c

```diff
@@ -61,11 +61,19 @@ static int do_lsmod(int argc, char *argv[])
 		const char *name = kmod_module_get_name(mod);
 		int use_count = kmod_module_get_refcnt(mod);
 		long size = kmod_module_get_size(mod);
-		struct kmod_list *holders, *hitr;
+		struct kmod_list *holders = kmod_module_get_holders(mod), *hitr;
 		int first = 1;
 
+		if (holders == NULL || use_count < 0 || size < 0) {
```

Comment:
holders == NULL  is a normal thing to have, e.g:

```
serio_raw              20480  0
```

`use_count < 0` allows us to use modules in the "going" state, i.e. in the process of being unloaded.

`size < 0` I believe is the only one that we should really do something.... maybe setting it to 0.  Rather than exiting here, I think we should just handle the races as a normal situation and continue printing the rest of the list.